storage/aptosdb/src/db/include/aptosdb_writer.rs

```diff
@@ -630,4 +745,42 @@ impl AptosDB {
 
         Ok(())
     }
+
+    // Update in-memory state of the database and the metrics before reverting.
+    // Note that any failures in persisting the revert should be treated as
+    // non-recoverable.
+    fn pre_revert(
```

Comment:
This is also a cherry-pick from #28, not yet used anywhere, but will be once that PR lands.